### PR TITLE
Hibernate 6 support

### PR DESCRIPTION
**Pull request by markusheiden**
Still waiting for spring-orm 6.0.0-M4 with Hibernate 6 support.

### CODE REVIEW

**Comment by arey** [comment]
Spring Framework is only used for unit testing. Thus using a Milestone version could be fine.

**Comment by markusheiden** [comment]
Currently the tests do not even compile and I haven't tested the code because my apps all use Spring.

What do you think about removing the Spring dependencies for tests?

**Comment by arey** [comment]
I'm not against removing the dependency to Spring Test. However, I'm no longer used to working without at the persistence layer level. How would you go about it?

**Comment by markusheiden** [comment]
Me too. But I would simply read the Hibernate docs. Technically I couldn't be that hard.

**Comment by markusheiden** [comment]
I managed to fix the code and all tests (hopefully) with Spring. Just one test seems to show a bug in Hibernate 6.

**Comment by markusheiden** [comment]
I reported https://hibernate.atlassian.net/browse/HHH-15713 to fix the issue with the broken test.

**Comment by markusheiden** [comment]
@arey I am ready now. May you please review this PR? Shall we wait for HHH-15713 to be fixed or is this change releasable?

File: src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java [hunk 1]
```diff
@@ -58,74 +57,82 @@ void setUp() {
 		dbUnitLoader.loadDatabase(getClass());
 	}
 
+	// TODO markus 2022-11-13: Re-enable when bug has been fixed in Hibernate.
+	@Disabled("Broken because of Hibernate bug HHH-15713")
```
Comment (arey) [inline src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java:61]: Maybe create an issue in the `hibernate-hydrate` in order to remember about this TODO.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -11,13 +11,14 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
-      - name: Set up JDK 1.8
-        uses: actions/setup-java@v1
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
         with:
-          java-version: 1.8
+          distribution: 'temurin'
+          java-version: '17'
```
Comment (arey) [inline .github/workflows/build.yml:21]: Does Hibernate ORM 6 only support Java 17?

File: readme.md [hunk 3]
```diff
@@ -25,7 +25,14 @@ A french article titled *Say goodbye to LazyInitializationException* : http://ja
 Download the jar though Maven:
 
 ```xml
-<!-- Either Hibernate 5.2 and above support -->
+<!-- Either Hibernate 6 and above support -->
+<dependency>
+  <groupId>com.javaetmoi.core</groupId>
+  <artifactId>javaetmoi-hibernate6-hydrate</artifactId>
```
Comment (arey) [inline readme.md:31]: Maybe rename the `artifactId` of the `pom.xml` to `javaetmoi-hibernate6-hydrate`

File: readme.md [hunk 4]
```diff
@@ -25,7 +25,14 @@ A french article titled *Say goodbye to LazyInitializationException* : http://ja
...
+  <version>6.0.0</version>
```
Comment (arey) [inline readme.md:32]: Maybe change the `version` of the `pom.xml` to `6.0.0-SNAPSHOT`

File: src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java [hunk 5]
```diff
@@ -58,74 +57,82 @@ void setUp() {
...
 
-			List<System> retrievedSystems = (List<System>) sessionFactory.getCurrentSession()
-					.createCriteria(System.class)
-					.addOrder(Order.asc("systemNumber")).list();
-			LazyLoadingUtil.deepHydrate(sessionFactory.getCurrentSession(), system);
-			return retrievedSystems;
+			selectAllSystemsOrderedByNumber(entityManager);
+			return LazyLoadingUtil.deepHydrate(entityManager, system);
 		});
+		// TODO markus 2022-11-06: Don't we need some assertions here?
```
Comment (arey) [inline src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java:127]: Yes we should compare the `dbContainer` list with the expected one.

File: src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java (hunk 1 above)
Comment (markusheiden) [reply src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java:61]: Created #34.

File: .github/workflows/build.yml (hunk 2 above)
Comment (markusheiden) [reply .github/workflows/build.yml:21]: No Hibernate 6 just needs Java. But Spring Boot 3 / Spring 6 needs JDK 17.

File: readme.md (hunk 4 above)
Comment (markusheiden) [reply readme.md:32]: Done.

File: src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java (hunk 5 above)
Comment (markusheiden) [reply src/test/java/com/javaetmoi/core/persistence/hibernate/TestIssue3.java]: I added some assertions. Please have a look at them.

**Comment by markusheiden** [comment]
Thanks for the review and the vote!

I fixed or commented all the issues. May you please have a second look? :-)

File: .github/workflows/build.yml (hunk 2 above)
Comment (arey) [reply .github/workflows/build.yml:21]: It's ok. But we still using Spring 5.3 (waiting the 6.0 release that will come soon)

**Comment by arey** [comment]
Hi @markusheiden,

The maven release has failed because I build it against Java 11 and Spring Framework 6 requires Java 17: https://github.com/arey/hibernate-hydrate/actions/runs/3477289438/jobs/5813300578#step:6:1805

I could upgrade the github actions file to Java 17 but it means that Hibernate Hydrate could not be used with Java 11.
I could also disable unit testing during the release process. We are using Spring only for testing I suppose.
Other solution: remove Spring usage

What is your suggestion?

**Comment by markusheiden** [comment]
Hi @arey,

Spring 6 is required because Spring 5 does not support Hibernate 6. So JDK 17 is required for compiling the tests.

Because we do not use Spring in the "production" code and because of 
```xml
   <maven.compiler.source>11</maven.compiler.source>
    <maven.compiler.target>11</maven.compiler.target>
```
in the pom.xml the libs should be usable with JDK 11.

**Comment by arey** [comment]
The maven release is successful. It should soon appear in Maven Central. 
Good job Markus :)

**Pull request merged**